### ISSUE TITLE
Use volume bibstring instead of jourvol in my bibliography

### ISSUE THREAD

**Issue by carepack** [michal-h21/biblatex-iso690#23]
hi,

you're work is amazing. a big help. One little thing. Why is the issue field not included? Is it possible to do it? How? Deadline fo my work is tomorrow :-(
Volume means the year of publishing. But I should print the issue number. Another idea would be that the volume field is not getting Jg (German for publishing year) in front. Instead Vol. would be perfect

thx a lot

**Comment by Kobzol** [comment]
What type of publication are you using? Issue only works with article and period AFAIK.

**Comment by carepack** [comment]
wow. thx for the fast reply. i found out it's a language problem. If my whole document is in english then
volume = {blabla} -> looks then Vol. blabla
When my document is in german it looks like this

volume = {blabla} -> looks then Jg. blabla

**Comment by johannesbottcher** [comment]
`\DefineBibliographyStrings{german}{volume={carepack}}`

Can you show us a minimal example after you are done that shows that behaviour?

**Comment by moewew** [comment]
Maybe I'm a bit thick this morning, but I'm not sure if I understand what you want.

Anyway, often people use the field `issue` while `number` is the more appropriate field. While the journal page might say "volume 33, issue 1", in `biblatex` speak here "1" is a `number`, `issue` is reserved for something with a name (roughly) like "Spring issue" or the like.

Currently for `volume` we have

```
\DeclareFieldFormat*{volume}{\bibsstring{jourvol}\space#1}
```
this adapts to the language of the entry if you want to stay with the main language use
```
\DeclareFieldFormat*{volume}{\mainsstring{jourvol}\space#1}
```

and if you are in a hurry go with
```
\DeclareFieldFormat*{volume}{Vol.~#1}
```

**Comment by johannesbottcher** [comment]
>Currently for volume we have

>`\DeclareFieldFormat*{volume}{\bibsstring{jourvol}\space#1}`


o.O Arent `volume` and `jourvol` two very different things?

On the other hand, following example is ... interesting.

```syntax=latex
\documentclass[ngerman,english]{article}
\usepackage{babel}
\usepackage[style=iso-authoryear]{biblatex}
\addbibresource{biblatex-examples.bib}
\begin{document}
\fullcite{knuth:ct:c}\par
\selectlanguage{ngerman}
\fullcite{knuth:ct:c}
\printbibliography
\end{document}
```


Having read in the iso in english and german (not czech) i really have to say, standard biblatex is doing just fine. So far, drivers are missing for music and stuff like that, but for usual journals/books everything is fine. The standard states, that they do not make any recommendations on punctuation.

**Comment by moewew** [comment]
@johannesbottcher  Yes, you are absolutely right there, we probably need something like

```
\DeclareFieldFormat{volume}{\bibsstring{volume}~#1}% volume of a book
\DeclareFieldFormat[article,periodical]{volume}{\bibsstring{jourvol}~#1}% volume of a journal
```
(there is also the issue of breakable `\space` vs. non-breaking `~`).

Unfortunately (or not), I have not yet had the pleasure of reading the original ISO/DIN documents about this, only the second-hand summaries on the internet, so I can't say anything about design decisions  and how well the standard styles implement them. (Many of the things I have seen claimed to be ISO/DIN are horrendous and frankly bonkers: change the language for some strings, but not all; spaces before colons. But then again I read - and you seem to confirm this - that there is a certain degree of freedom in the recommendation - which doesn't preclude Professors from only accepting a very weirdly punctuated bibliography as really ISO/DIN compliant.) Currently all I can do is streamline the status quo and implement small new suggestions. I would be interested in helping to make this style here fully ISO/DIN compliant with as few changes from standard `biblatex` as possible, but I'm probably not going to find the time in the next six months.

**Comment by johannesbottcher** [comment]
The problem is that the norm is expensive. University members can often get a copy using the library. 

http://chat.stackexchange.com/transcript/41?m=27447418#27447418

The norm defines what author to print first (leading author, who would have guessed) or what happens if the author is an organization (use the org as the author). What happens with name prefixes. What about online entries? Try to figure out if there is an author, place URL and access date.

So this, with respect to biblatex, is more or less about *Database Input*. For instance, all things discussed at the point are typeset IN CAPITALS in the examples. This is the highlighting decision the standard chose. It doesn't mean (imho) that authors should be printed uppercased. I refer to the linked chat message, a direct quote from the standard.

I did not check the czech standard, because i don't read/speak the language, but the main point of the german and english standard are: Be consistent and clear to make the items findable.

**Comment by moewew** [comment]
Interesting point, I thought they would at least impose an order of the (`biblatex`) fields.

Indeed our university library has the norm (but only in the 1995 version), so once I have the time, I'll be sure it give it a read.

Anyway, I forgot to comment on your second observation. I think this has to do with `\frenchspacing`/`\nonfrenchspacing` and is present with other styles as well, see

```
\documentclass[ngerman,english]{article}
\usepackage{babel}
\usepackage[style=authoryear]{biblatex}
\addbibresource{biblatex-examples.bib}
\begin{document}
\fullcite{knuth:ct:c}\par
\selectlanguage{ngerman}
\fullcite{knuth:ct:c}
\printbibliography
\end{document}
```

though the effect is much more subtle here; and

```
\documentclass[ngerman,english]{article}
\usepackage{babel}
\usepackage[style=iso-authoryear]{biblatex}
\addbibresource{biblatex-examples.bib}
\begin{document}
\frenchspacing
\fullcite{knuth:ct:c}\par
\selectlanguage{ngerman}
\fullcite{knuth:ct:c}
\printbibliography
\end{document}
```

**Comment by michal-h21** [comment]
I have access to the 2010 version of the norm and it isn't really informative. It states that the most important thing is to be consistent (which is logical :), but the examples are rather ambiguous: Journal volume can be cited as vol. 4, n. 2 or **4**(2), in author list it states that first author should be in inverted form and others in natural order **if necessary** (I found this form absolutely horrible, so I conclude that it isn't necessary and keep the form from the older ISO version). The `\mainbibstring` vs. `\bibstring` business is really included in the norm, as well as request to transliterate non-latin strings. 

The Czech interpretation, which is imposed by many Czech universities is much more strict than the actual norm. Capital authors are required for example. 

The spacing issue is probably due to babel switching `\frenchspacing`, do you have idea why it there is much more space in the ISO sample?

Good catch about `volume`, this field is used actually only in the `article` entry type. It would be really nice if we could just use some modifications for standard `biblatex` styles. most of the code is relic from earlier `biblatex` versions, total rewrite would be probably easier than adapting it to the current `biblatex`. But I don't have knowledge of the current `biblatex` internals and also not much time for this :(

**Comment by carepack** [comment]
to all of you amazing guys: thank you. you saved my a**. I've uploaded the paper few minutes ago. That's why I haven't answered earlier. I tried the field remapping suggested by moewew.
\DeclareFieldFormat*{volume}{Vol.~#1} does the (fast) trick.

I want to explain my problem anyway. Maybe useful for others.
a bibliography entry for example:

@ARTICLE{fallgatter2002theorie,
author = {Fallgatter, Michael},
title = {Theorie des Entrepreneurships: Perspektiven zur Erforschung der Entstehung und Entwicklung junger Unternehmungen},
journal = {neue betriebswirtschaftliche forschung},
year = {2002},
volume = {299}
}

the expected result should like like this for me. It fits for me and the formal criteria of my university best.

FALLGATTER, Michael, 2002. Theorie des Entrepreneurships: Perspektiven zur Er-
forschung der Entstehung und Entwicklung junger Unternehmungen. neue be-
triebswirtschaftliche forschung, **Vol. 299**.

If the volume has a number I can add them too. That's not the problem. Before adding moewew command the result looked like this:


FALLGATTER, Michael, 2002. Theorie des Entrepreneurships: Perspektiven zur Er-
forschung der Entstehung und Entwicklung junger Unternehmungen. neue be-
triebswirtschaftliche forschung, **Jg. 299**.

Jg in german is the year of publishing the journal. Of yourse I can bypass this by using the number fiel instead the volume. But I have some articles in my work they need two field. Vol. 299 Nr. 4
for example. Hope this make things clearer now.

Oh man I would like to buy a coffee to all of you ;)
 big thx again. And yes I was in hurry ;)
